features/util.py

```diff
@@ -84,9 +88,19 @@ def lxc_get_property(name: str, property_name: str, image: bool = False):
         return value
 
 
-def build_debs(
-    series: str, output_deb_dir: str, cache_source: bool
-) -> List[str]:
+def repo_state_hash(include_features: bool = False):
+    git_log_cmd = ["git", "log", "-1"]
+    git_diff_cmd = ["git", "diff"]
+    if not include_features:
+        files_and_folders = [f for f in os.listdir(".") if "features" not in f]
+        git_diff_cmd += ["--", *files_and_folders]
+    log_out = subprocess.check_output(git_log_cmd)
+    diff_out = subprocess.check_output(git_diff_cmd)
+    output_to_hash = log_out + diff_out
+    return hashlib.md5(output_to_hash).hexdigest()
```

Comment:
This is the interesting bit. By hashing the text that includes the current commit and the current diff, we get a string that represents the current state of the local repo, including the dirty bits.

The other code is to exclude the `features` directory from this, so as not to rebuild when you're iterating on an integration test

Reply:
I think this approach will work fine, we always look at the most recent commit and check for any new diffs in the files. I think we should only document this approach for future interaction, but it is a great idea

Reply:
Good point, I will add a docstring to this function